### PR TITLE
ISSUE-138: Make json libraries pluggable for blacksheep

### PR DESCRIPTION
**Pull request by Cdayz**
Fixes issue #138

### CODE REVIEW

**Comment by RobertoPrevato** [comment]
Thanks, Nikita @Cdayz - this looks great! I leave only three small comments.

File: itests/app_orjson.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    def default(x):
+        if isinstance(x, bytes):
+            return base64.urlsafe_b64encode(x).decode('utf-8')
+
+        raise TypeError
+
+    return orjson.dumps(obj, default=default).decode('utf-8')
+
+
+app_orjson = Application(
```
Comment (Cdayz) [inline itests/app_orjson.py:23]: Hi, a made some fixes, and re-write plugins to use them as non-global objects, they now can be configured in such way

File: tests/test_responses.py [hunk 2]
```diff
@@ -905,7 +905,9 @@ def __init__(self) -> None:
             self.might_be_secret = "iunisud109283012"
 
     with pytest.raises(TypeError):
-        json(NotSerializable())
+        response = json(NotSerializable())
+        # NOTE: it's a trade-off to serialize at content level instead of serialize at json function call
```
Comment (Cdayz) [inline tests/test_responses.py:909]: But, after re-writing i changed one of tests, because now serialization to json can be proceeded only when `response.content.body` is accesed

File: itests/app_orjson.py (hunk 1 above)
Comment (RobertoPrevato) [reply itests/app_orjson.py:23]: Hi! Thanks again for your contribution - after reviewing your latest changes, I decided to try an alternative way. I didn't want to annoy you asking to try another thing, so I forked your fork and tried myself

1. I confirm that with the code you wrote initially, the whole test suite was running with specific JSON settings using `orjson`, not only the application whose variable is named `app_orjson`
2. to keep the JSON settings global, which is fine for regular users of the framework and only not fine in the context of the test suite for BlackSheep, I leveraged instead the principle of process forking, so I slightly modified your code to apply specific JSON settings only in the process that runs `app_orjson`

In other words, instead of making many changes to the existing code to support non-global JSON settings, I tried to apply specific JSON settings only to the new tests you added. I like this solution more.

I will create a second PR, where we will result both as contributors. 😄

**Comment by RobertoPrevato** [comment]
Hi @Cdayz, I created a second PR with your changes + the things I described above. I also used `black` and `isort` to apply the same formatting and sort import statements.
